In [3]:
import pandas as pd
import numpy as np


df = pd.read_csv('data/heart.csv')


corr_matrix = df.corr()

upper = corr_matrix.where(
    np.triu(np.ones(corr_matrix.shape), k=1).astype(bool)
)

corr_pairs = upper.unstack()

corr_pairs = corr_pairs.dropna()

corr_pairs = corr_pairs.sort_values(key=abs, ascending=False)

top_3 = corr_pairs.head(3)

print(top_3)

slope   oldpeak   -0.577537
target  exang     -0.436757
        cp         0.433798
dtype: float64


task034 kiana


Slope_oldpeak = -0.57
Slopeوoldpeak رابطه عکس قوی ای دارن یعنی با افزایش oldpeak , slope کاهش پیدا می کنه پس این دو ویژگی بهم وابسته هستند 


target_exang = -0.43
رابطه عکس تقریباً قوی ای دارن با همدیگه چون منفیه  و مثل بالایی با بیشتر شدن یکی اون یکی کاهش پیدا می کنه 

target_cp = 0.43
رابطه مستقیم قوی دارن و اگه یکی افزایش پیدا کنه اون یکی هم افزایش پیدا می کنه و کاملا بهم وابسته هستند